services/src/main/java/org/keycloak/services/resources/admin/UsersResource.java

```diff
@@ -150,6 +152,8 @@ public Response createUser(final UserRepresentation rep) {
         }
 
         try {
+            UserEventContext userEvent = UserEvents.createByAdmin(session, rep).firePreEvent();
```

Comment:
Should this really be fired in the Ressource. Should not this be implemented in the UserStorageProvider or UserStorageManager? 

I am just wondering because there may be use cases where creation or deletion of a user may not be triggered by an API. For example, UserStorageManager has a feature 'import validation' which deletes users from Keycloak when they are invalid or do not exist any longern in the system integrated by the federation provider.